lib/sygus_converter.js

```diff
@@ -45,51 +74,101 @@ export function extractFunctions(code) {
   var sygusBody = jsToSygus(fxn.body);
   return "(define-fun "+fxn.name+" ("+argNamesSygus+")"+returnType+sygusBody+")";
 }
 
 function genFunctionSynth(fxn, intendedExamples) {
-  var argNamesSygus = genParams(fxn);
+  var types = getInputOutputType(intendedExamples, fxn);
+  var paramTypes = types[0];
+  var returnType = types[1];
+  //console.log(returnType);
+  var argNamesSygus = genParams(fxn, paramTypes);
   //TODO get actual return type
-  var synthCommand = "(synth-fun "+fxn.name+" ("+argNamesSygus+") String \n";
-  return synthCommand + genGrammar(fxn.params, intendedExamples);
+  var synthCommand = "(synth-fun "+fxn.name+" ("+argNamesSygus+") "+returnType+ " \n";
+  return synthCommand + genGrammar(fxn.params, intendedExamples, paramTypes, returnType);
+};
+
+function getType(object){
+  return typeof object;
 };
 
 //TODO select grammar elements based on user code
-function genGrammar(params, intendedExamples) {
-  var exConsts = new Set();
+//TODO edit to allow for different types
+function genGrammar(params, intendedExamples, inTypes, outType) {
+  var consts = {};
   for (i=0; i < intendedExamples.length; i+=1) {
+    if (typeof intendedExamples[i] == "string"){
     //exConsts.push(JSON.stringify(intendedExamples[i]));
     for (j=intendedExamples[i].indexOf("(")+1; j < intendedExamples[i].length; j++) {
       if (intendedExamples[i][j]!="\"") {
-         exConsts.add(JSON.stringify(intendedExamples[i][j]));
+         if (!isNaN(parseInt(intendedExamples[i][j]))) {
+         if ((Object.keys(consts)).includes("number")){
+           consts["number"].add(String(parseInt(intendedExamples[i][j])));
+         } else{
+           consts["number"] = new Set(String(parseInt(intendedExamples[i][j])));
+
+         }
+
+       } else {
+         var type = getType(intendedExamples[i][j]);
+         if ((Object.keys(consts)).includes(type)){
+           consts[type].add(JSON.stringify(intendedExamples[i][j]));
+         } else {
+           consts[type] = new Set();
+           consts[type].add(JSON.stringify(intendedExamples[i][j]))
+         }
+
       }
+
+     }
     }
   }
-  return (`((Start String (ntString))
+  }
+
+  if (!((Object.keys(consts)).includes("number"))) {
+    consts["number"] = new Set();
+  }
+  for (i = 0; i<=5; i++){
+      consts["number"].add(i);
+  }
+  if (!((Object.keys(consts)).includes("boolean"))) {
+    consts["boolean"] = new Set();
+  }
+  consts["boolean"].add("true");
+  consts["boolean"].add("false");
+
+  consts["number"] = new Set(Array.from(consts["number"]).filter((v, i, a) => a.indexOf(v) == i));
+  consts["string"] = new Set(Array.from(consts["string"]).filter((v, i, a) => a.indexOf(v) == i));
+
+
+  //console.log(Array.from(consts["string"]));
+  return (`((Start `+outType+` (nt`+outType+`))
 (ntString String (`+
-  (params.map(c => c.name)).join(" ") + " " +
-  (Array.from(exConsts).join(" ")) +
+  ((params.map(c => c.name)).filter(param => inTypes[param] == "String")).join(" ") + " " +
```

Comment:
these two lines are repeated three times, with the only change being to the type. Maybe this could be a function? Something like ```genConstantsOfType(type)```